lib/auth/auth_with_roles.go

```diff
@@ -3125,6 +3125,20 @@ func (a *ServerWithRoles) GetSessionEvents(namespace string, sid session.ID, aft
 		return nil, trace.Wrap(err)
 	}
 
+	// emit a session recording view event for the audit log
+	if err := a.authServer.emitter.EmitAuditEvent(a.authServer.closeCtx, &apievents.SessionRecordingAccess{
+		Metadata: apievents.Metadata{
+			Type: events.SessionRecordingAccessEvent,
+			Code: events.SessionRecordingAccessCode,
+		},
+		SessionID: sid.String(),
+		UserMetadata: apievents.UserMetadata{
+			User: a.context.User.GetName(),
+		},
```

Comment:
Can we use `ClientUserMetadata(ctx)` instead?

Reply:
Does that work with `a.context`? I only see it used once in `auth_with_roles.go`, and that `ctx` is passed into the method, whereas `GetSessionEvents` doesn't get passed a `ctx` - just want to make sure they're effectively the same context?

Reply:
Looks like `a.context.Identity.GetIdentity().GetUserMetadata()` will do it, we have the `Identity` easily accessible in `*ServerWithRoles`.